File: WorkoutTracker.Domain/Entities/Exercises/ExerciseRecords/ExerciseRecordGroup.cs
using WorkoutTracker.Domain.Base;
using WorkoutTracker.Domain.Entities.Workouts;

namespace WorkoutTracker.Domain.Entities.Exercises.ExerciseGroups;

public class ExerciseRecordGroup : IDbModel
{
    public long Id { get; set; }

    public long ExerciseId { get; set; }
    public Exercise? Exercise { get; set; }

    public long WorkoutRecordId { get; set; }
    public WorkoutRecord? WorkoutRecord { get; set; }

    public IEnumerable<ExerciseRecord> ExerciseRecords { get; set; } = null!;
}

File: WorkoutTracker/src/app/shared/models/exercises/exercise-record-group.ts
import { DbModel } from "src/app/shared/models/base/db-model";
import { ExerciseType } from "src/app/exercises/models/exercise-type";
import { ModelWeight } from "../model-weight";
import { ExerciseRecord } from "src/app/exercises/models/exercise-record";

export interface ExerciseRecordGroup extends DbModel {
    sets: number;
    weight: ModelWeight;

    exerciseId: number;
    exerciseName: string;
    exerciseType: ExerciseType;

    exerciseRecords: ExerciseRecord[];
}